### DISCUSSION TITLE
Protobuf support

### DISCUSSION THREAD

**Discussion by dmytro-babichev** [instancio/instancio#604] [Ideas]
Hey all. First of all thanks for the great job! I adore the Instancio framework. Very cool stuff!

However, I faced with an issue which blocks me from using it on my project. We use Protobuf quite extensively and it appears that Instancio is not compatible with classes and builders that `protoc` generates.

I tried debugging it a bit, but it seems the problem is related to how Protobuf classes store their fields. Most of the fields are of type `Object` and type safety is achieved by the implementation of setters and getters. Also there are a few "internal" fields that must not be populated by a client.

I'm not sure if can be implemented easily, but it would be fantastic to be able to generate protobuf messages for testing using Instancio.

What do you think?

**Comment by armandino** [comment]
@dmytro-babichev thank you very much for the feedback! It means a lot :)

I agree that it'd be great to add support for protobufs. I would need to look into it more closely to see how feasible it is to implement. I created an issue #606 to investigate it further.

**Reply by armandino** [reply]
The new `instancio-protobuf` module is now available in `6.0.0-RC3` 

https://central.sonatype.com/artifact/org.instancio/instancio-protobuf
